DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DelegateCoursesController.cs

```diff
@@ -0,0 +1,101 @@
+﻿namespace DigitalLearningSolutions.Web.Controllers.TrackingSystem.Delegates
+{
+    using System.Linq;
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Enums;
+    using DigitalLearningSolutions.Data.Helpers;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Attributes;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.Helpers.FilterOptions;
+    using DigitalLearningSolutions.Web.Models.Enums;
+    using DigitalLearningSolutions.Web.ViewModels.TrackingSystem.CourseSetup;
+    using DigitalLearningSolutions.Web.ViewModels.TrackingSystem.Delegates.DelegateCourses;
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.FeatureManagement.Mvc;
+
+    [FeatureGate(FeatureFlags.RefactoredTrackingSystem)]
+    [Authorize(Policy = CustomPolicies.UserCentreAdmin)]
+    [SetDlsSubApplication(nameof(DlsSubApplication.TrackingSystem))]
+    [SetSelectedTab(nameof(NavMenuTab.Delegates))]
+    [Route("TrackingSystem/Delegates/Courses")]
+    public class DelegateCoursesController : Controller
+    {
+        private const string CourseFilterCookieName = "DelegateCoursesFilter";
+        private readonly ICourseCategoriesDataService courseCategoriesDataService;
+        private readonly ICourseService courseService;
+        private readonly ICourseTopicsDataService courseTopicsDataService;
+
+        public DelegateCoursesController(
+            ICourseService courseService,
+            ICourseCategoriesDataService courseCategoriesDataService,
+            ICourseTopicsDataService courseTopicsDataService
+        )
+        {
+            this.courseService = courseService;
+            this.courseCategoriesDataService = courseCategoriesDataService;
+            this.courseTopicsDataService = courseTopicsDataService;
+        }
+
+        [Route("{page=1:int}")]
+        public IActionResult Index(
+            string? searchString = null,
+            string? sortBy = null,
+            string sortDirection = GenericSortingHelper.Ascending,
+            string? filterBy = null,
+            string? filterValue = null,
+            int page = 1,
+            int? itemsPerPage = null
```

Comment:
We haven't had a request to set items per page here. Let's check if they want it.
Kevin's away today and tomorrow, but could you leave a question for David and Carolyn (probably include Kevin for when he gets back) on #hee-dls-refactor asking if they want to be able to set the number of courses per page on Delegate courses?

Reply:
If they don't want it, there's a _SearchResultsCount.cshtml partial which doesn't include an `ItemsPerPage` dropdown.